File: server_old/src/shared/models/entities/post/post.ts
import {WbEntity} from '../wb-entity';
import {
	Column,
	Entity,
	ManyToOne,
	OneToMany
} from 'typeorm';
import {User} from '../user/user';
import {Comment} from '../comment/comment';

@Entity()
export class Post extends WbEntity {
	@Column({type: 'text'})
	content: string;
	
	@Column({type: 'text', nullable: true})
	file: string;
	
	@ManyToOne(
			() => User,
			user => user.posts,
	)
	user: User;
	
	@OneToMany(
			() => Comment,
			comment => comment.post,
			{
				nullable: true,
				cascade: true,
			}
	)
	comments: Comment[];
}

File: server/src/main/kotlin/br/com/webbicho/server/shared/entities/Post.kt
package br.com.webbicho.server.shared.entities

import jakarta.persistence.*
import java.util.*

@Entity(name = "posts")
class Post(
	@Column(columnDefinition = "TEXT", nullable = false)
	var content: String,
	
	// Implements files support
	
	@Column(name = "user_uuid", nullable = false)
	var userUuid: UUID,
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(
		name = "user_uuid",
		nullable = false,
		insertable = false,
		updatable = false,
	)
	private var user: User? = null,
	
	@OneToMany(
		mappedBy = "post",
		cascade = [CascadeType.REMOVE],
		orphanRemoval = true,
		fetch = FetchType.LAZY
	)
	private val comments: MutableList<Comment> = mutableListOf()
) : WbEntity()
